### ISSUE TITLE
rtiddsgen crashes sporadically on OSX

### ISSUE THREAD

**Issue by mikaelarguedas** [ros2/build_farmer#40]
we knew that was an issue requiring patching on Windows. It showed up on  OSX last night:
at elapsed time 00:27:34.721: http://ci.ros2.org/view/nightly/job/nightly_osx_debug/514.

I remember seeing the same happen on Linux recently, not sure if it was my work station or a CI job.

**Comment by mikaelarguedas** [comment]
Error on Linux workstation when building in parallel

Linux error moved to https://github.com/ros2/build_cop/issues/47

**Comment by wjwwood** [comment]
I think the Linux issue in @mikaelarguedas's comment is a different issue. He's going to make a new issue for that.

But there is a new instance of this issue which I reported in #46 too.

```
08:00:44 subprocess.CalledProcessError: Command '['/Applications/rti_connext_dds-5.2.3/bin/rtiddsgen_server', '-I', '/Users/osrf/jenkins/workspace/nightly_osx_release/ws/build/nav_msgs/rosidl_generator_dds_idl', '-I', '/Users/osrf/jenkins/workspace/nightly_osx_release/ws/install/builtin_interfaces/share', '-I', '/Users/osrf/jenkins/workspace/nightly_osx_release/ws/install/geometry_msgs/share', '-I', '/Users/osrf/jenkins/workspace/nightly_osx_release/ws/install/std_msgs/share', '-d', '/Users/osrf/jenkins/workspace/nightly_osx_release/ws/build/nav_msgs/rosidl_typesupport_connext_cpp/nav_msgs/srv/dds_connext', '-language', 'C++', '-namespace', '-update', 'typefiles', '-unboundedSupport', '/Users/osrf/jenkins/workspace/nightly_osx_release/ws/build/nav_msgs/rosidl_generator_dds_idl/nav_msgs/srv/dds_connext/GetMap_Request_.idl']' died with <Signals.SIGSEGV: 11>.
```
http://ci.ros2.org/job/nightly_osx_release/570/consoleFull

**Comment by dirk-thomas** [comment]
Please consider reporting this to RTI together with a reproducible example if that is still a problem in Connext 5.3.

**Comment by mikaelarguedas** [comment]
Given that it's a sporadic issue we would need to be able to run a bunch of osx jobs with 5.3 to confirm/infirm this is still an issue. Do we have a way to do this currently ?

**Comment by dirk-thomas** [comment]
You need to download the latest version for OS X and install it on one of the nodes.

**Comment by dirk-thomas** [comment]
I will close this ticket for now since I am not aware of this happening since we moved to Connext 5.3.

**Comment by dhood** [comment]
This happened again last night on Windows:
https://ci.ros2.org/view/nightly/job/nightly_win_rel/747/consoleFull
```
23:48:57          Creating "x64\Release\rcl_interfaces__dds_connext_idl\rcl_inte.8597C808.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
23:48:57        CustomBuild:
23:48:57          Building Custom Rule C:/J/workspace/nightly_win_rel/ws/src/ros2/rcl_interfaces/rcl_interfaces/CMakeLists.txt
23:48:58          CMake does not need to re-run because C:/J/workspace/nightly_win_rel/ws/build/rcl_interfaces/CMakeFiles/generate.stamp is up-to-date.
23:48:58          Generating DDS interfaces
23:48:58          Generating C++ type support for RTI Connext (using 'rtiddsgen_server.bat')
23:48:58   Duration_Support.cxx
23:48:58     27>C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets(171,5): error MSB6006: "cmd.exe" exited with code -1073741819. 
```

**Comment by dhood** [comment]
that's with parallel outputting, so I am not completely certain that ddsgen is what caused it, but it certainly wouldn't be surprising. From speaking with @mikaelarguedas it sounds like this shouldn't be necessary with 5.3 anymore. What's the communication channel with them to report that this is still an issue?

**Comment by mikaelarguedas** [comment]
for context: in that sentence: `it sounds like this shouldn't be necessary with 5.3 anymore`, *this* refers to the patched version of `rtiddsgen` we were using in 5.2.3

**Comment by dirk-thomas** [comment]
First of all the problem needs to be clearly identified that the generator crashed (which seems to be not yet the case?). Then I think it needs a reproducible example. That information can then be passed to the normal RTI support channels: support@rti.com

**Comment by dirk-thomas** [comment]
I don't think the title of an already closed ticket should be changed like this. The original problem was reported for OS X. Changing it to be Windows specific now will make the issue undiscoverable if we ever try to search for the OS X specific problem.

**Comment by dhood** [comment]
Yep, makes sense. spun off in https://github.com/ros2/build_cop/issues/95
